src/components/SafeListSidebar/SafeList/index.tsx

```diff
@@ -46,70 +42,70 @@ const useStyles = makeStyles({
-    return address === currentSafeAddress && !isAddressAdded(safes, address)
-  })
-
-  const getLink = (address: string): React.ReactElement =>
-    sameAddress(currentSafeAddress, address) ? (
-      <StyledIcon type="check" size="md" color="primary" />
-    ) : (
-      <div className={classes.noIcon}>placeholder</div>
-    )
+  const networks = getNetworks()
+  const loadedSafes = useSelector(sortedSafeListSelector).filter(isNotLoadedViaUrl)
+  const ownedSafes = useOwnerSafes()
+  const localSafes = useLocalSafes()
 
   return (
-    <MuiList className={classes.list}>
-      {safes.map((safe) => (
-        <Fragment key={safe.address}>
-          <Link
-            data-testid={SIDEBAR_SAFELIST_ROW_TESTID}
-            onClick={onSafeClick}
-            to={generatePath(SAFE_ROUTES.ASSETS_BALANCES, {
-              safeAddress: safe.address,
-            })}
-          >
-            <ListItem classes={{ root: classes.listItemRoot }}>
-              {getLink(safe.address)}
-              <AddressWrapper safe={safe} />
-            </ListItem>
-          </Link>
-          <Hairline />
-        </Fragment>
-      ))}
+    <StyledList>
+      {networks.map(({ id, backgroundColor, textColor, label }) => {
+        const isCurrentNetwork = id === getNetworkId()
+
+        const localSafesOnNetwork = localSafes[id].filter(isNotLoadedViaUrl)
+        const addedSafesOnNetwork = isCurrentNetwork ? loadedSafes : localSafesOnNetwork
+        const shouldExpandOwnedSafes = localSafesOnNetwork.some(({ address }) => isSafeAdded(loadedSafes, address))
 
-      {ownedSafes.length > 0 && (
-        <ListItem classes={{ root: classes.listItemCollapse }}>
-          <Collapse title={`All owned Safes (${ownedSafes.length})`} defaultExpanded={ownedSafesExpanded}>
-            {ownedSafes.map((address: string) => (
-              <Fragment key={address}>
-                <OwnedAddress address={address} onClick={onSafeClick} isAdded={isAddressAdded(safes, address)}>
-                  {getLink(address)}
-                </OwnedAddress>
-                <Hairline />
-              </Fragment>
-            ))}
-          </Collapse>
-        </ListItem>
-      )}
-    </MuiList>
+        if (!isCurrentNetwork && !localSafesOnNetwork.length && !addedSafesOnNetwork.length) return null
+        return (
+          <Fragment key={id}>
+            <ListItem selected>
+              <StyledDot backgroundColor={backgroundColor} textColor={textColor} />
+              {label}
+            </ListItem>
+            <MuiList>
+              {addedSafesOnNetwork.map((safe) => (
+                <SafeListItem
+                  key={safe.address}
+                  networkId={id}
+                  onNetworkSwitch={() => setNetwork(id)}
+                  onSafeClick={onSafeClick}
+                  safes={loadedSafes}
+                  {...safe}
+                />
+              ))}
+              {(ownedSafes.length > 0 || isCurrentNetwork) && (
```

Comment:
We don't want to show this menu if there are no owned safes.

To summarize:

* If there are added safes, show them
* If there are owned safes on the current network, show them
* If there are neither added nor owned safes, hide the network section completely

Reply:
![image](https://user-images.githubusercontent.com/20442784/134546988-c8e7bdf7-12b2-4b80-9610-10b425ced705.png)

Here's a screenshot of my most recent commit